### PR TITLE
Add week enhancement

### LINKED ISSUE #236: [Enhancement] Display weeks number in lesson information
## Current implementation

Currently, commands which show module information such as `showmod` and `listmod` only show lesson day and start time, it does not show the specific weeks the lessons occur on.

## Possible implementation

If a lesson occurs every week, no change to lesson display. However, if not display specific weeks which lessons occur on.

### PR DESCRIPTION
**Pull request by T-Wan-Lin**
Closes #236

### CODE REVIEW

File: src/main/java/seedu/apollo/module/Timetable.java [hunk 1]
```diff
@@ -61,6 +61,65 @@ public ArrayList<Integer> getWeeks() {
...
+        String formattedWeeks;
+
+        int firstWeek = unfilteredWeeks.get(0);
+        int lastWeek = unfilteredWeeks.get(unfilteredWeeks.size() - 1);
+
+        //First, check if the lesson spans the whole 13-week semester
+        if (unfilteredWeeks.size() == 13) {
+            formattedWeeks = "[Weekly]";
+            return formattedWeeks;
+        } else if (lastWeek - firstWeek == unfilteredWeeks.size() - 1) {
```
Comment (irving11119) [inline src/main/java/seedu/apollo/module/Timetable.java:87]: You can use if here instead of else if since the above if statement already returns something

**Review by irving11119 — COMMENTED** [review]
Just some minor CS1010esque things

File: src/main/java/seedu/apollo/module/Timetable.java [hunk 2]
```diff
@@ -61,6 +61,65 @@ public ArrayList<Integer> getWeeks() {
...
+        ArrayList<Integer> filteredWeeks = new ArrayList<>();
+        String formattedWeeks;
+
+        int firstWeek = unfilteredWeeks.get(0);
+        int lastWeek = unfilteredWeeks.get(unfilteredWeeks.size() - 1);
+
+        //First, check if the lesson spans the whole 13-week semester
+        if (unfilteredWeeks.size() == 13) {
+            formattedWeeks = "[Weekly]";
+            return formattedWeeks;
```
Comment (honglinshang) [inline src/main/java/seedu/apollo/module/Timetable.java:86]: oh actl if u want to nitpick for efficiency, u can check this immediately after line 76 and `return "[Weekly]"` if true to skip doing all the firstWeek lastWeek stuff

File: src/main/java/seedu/apollo/module/Timetable.java [hunk 3]
```diff
@@ -61,6 +61,65 @@ public ArrayList<Integer> getWeeks() {
...
+     * For example, if a lesson occurs on weeks 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, then
+     * the method will return [Weekly]. If a lesson does not occur for consecutive weeks, then
+     * it will print out all weeks that the lesson occurs on. For example, if a lesson occurs on
+     * weeks 5, 7, 9, 11, then the method will return [Weeks: 5, 7, 9, 11, 13]. Lastly if a lesson occurs on
+     * consecutive weeks but does not span the entire semester, then the method will return the first and last weeks.
+     * For example, if a lesson occurs on weeks 1, 2, 3, 4, 5, then the method will return [Weeks: 1-5].
+     * @param   timetable       The timetable of the lesson.
+     * @return  The formatted String representation of the weeks.
+     */
+    public String compressedWeeks(Timetable timetable) {
```
Comment (PoobalanAatmikaLakshmi) [inline src/main/java/seedu/apollo/module/Timetable.java:69]: Perhaps you could check that timetable is not null or thrown an exception if it is?

File: src/main/java/seedu/apollo/ui/Ui.java [hunk 4]
```diff
@@ -329,7 +330,7 @@ public void printSpecificTimetable(Module module, LessonType lessonType, ArrayLi
         for (Timetable timetable : copyList) {
             System.out.println("Class Number: " + timetable.getClassNumber());
             System.out.println("   " + timetable.getDay() + " " + timetable.getStartTime() + " - " +
-                    timetable.getEndTime());
+                                timetable.getEndTime() + " " + timetable.compressedWeeks(timetable));
```
Comment (yixuann02) [inline src/main/java/seedu/apollo/ui/Ui.java:333]: I think the indentation for this line is off, please check!

**Review by yixuann02 — COMMENTED** [review]
Minor code quality issues, overall good job.

File: src/main/java/seedu/apollo/ui/Ui.java (hunk 4 above)
Comment (T-Wan-Lin) [reply src/main/java/seedu/apollo/ui/Ui.java:333]: It's right on IntelliJ

**Pull request merged**